storybookjs/storybook
# Addon A11y: Avoid CSP issue

Closes #24355 






## What I did


Ensure Ace result is a simple json object that doesn't contain named objects, because telejson will deserialize it in a way that violates: `Content Security Policy directive: "script-src 'self' 'unsafe-inline'"`

There may be a better way to handle this, such as just converting the `CheckResult` objects to a simple object, but I think all the `Results` object can be stringified, so I went with a simple naive approach: `JSON.parse(JSON.stringify(result))`

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

I don't know if there is a good way to automate testing this. Maybe adding the CSP to an e2e test that runs `addon-a11y`. I can try to do that, if it would work, but I am not sure how, yet.

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_



1. Create a sandbox.
2. Create a `.storybook/manager-head.html` containing:
```html
<meta
  http-equiv="Content-Security-Policy"
  content="script-src 'self' 'unsafe-inline'"
/>
```
3. Run the sandbox and check the devtools for a CSP error.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
